src/storage-manager.coffee

```diff
@@ -67,6 +61,9 @@ class _StorageManager
     collection.on 'remove', (model) ->
       model.invalidateCache()
 
+    collection.on 'change', (model) =>
+      @_invalidateCache(model.brainstemKey) unless model.isNew()
```

Comment:
Ah so to add context, this cache should be used for `add`, `remove` and `reset` events which happen when pagination occurs through autocompleter loads or switching b/w actual pages.